### PR TITLE
add asdf-unit-schemas recipe

### PR DESCRIPTION
**Pull request by braingram**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by braingram** [comment]
@WilliamJamieson do you agree to be a maintainer?

File: recipes/asdf-unit-schemas/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/asdf_unit_schemas-{{ version }}.tar.gz
+  sha256: 42b78d67213efe4ffd4529fb0e58d9c7a0dab5cbf8839b230f1bc0a446bff999
+
+build:
+  skip: true  # [py<38]
```
Comment (WilliamJamieson) [inline recipes/asdf-unit-schemas/meta.yaml:13]:
```suggestion
  noarch: python
```

File: recipes/asdf-unit-schemas/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
```
Comment (WilliamJamieson) [inline recipes/asdf-unit-schemas/meta.yaml:22]:
```suggestion
    - python >=3.8
```

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]
Let's convert this to a `noarch` build.

 The following changes should do that

**Comment by WilliamJamieson** [comment]
I agree to be a maintainer.

File: recipes/asdf-unit-schemas/meta.yaml (hunk 1 above)
Comment (braingram) [reply recipes/asdf-unit-schemas/meta.yaml:13]: The linter complained that selectors cannot be used for noarch packages. Is there a way to handle the version dependency for importlib-resources without using a selector?
With [this version](https://github.com/conda-forge/staged-recipes/blob/ddefe581fcdf8a8f7d5acfe8159c4c87790c87f2/recipes/asdf-unit-schemas/meta.yaml) the linter reported: noarch packages can't have selectors. If the selectors are necessary, please remove noarch: python.
(in addition to errors for the python version having spaces).

Comment (dopplershift) [reply recipes/asdf-unit-schemas/meta.yaml:13]: We generally just unconditionally install things like importlib-resources in this case.

File: recipes/asdf-unit-schemas/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    - importlib-resources >=3  # [py<39]
```
Comment (WilliamJamieson) [inline recipes/asdf-unit-schemas/meta.yaml:23]:
```suggestion
    - importlib-resources >=3
```

This should not be necessary

**Pull request merged**